File: api/database/mongodb.js
import dotenv from "dotenv";
import { MongoClient } from "mongodb";

dotenv.config();

const mongoEnv = process.env.MONGO_ENV || "local";
const uri =
  mongoEnv === "atlas"
    ? process.env.MONGODB_URI_ATLAS
    : process.env.MONGODB_URI_LOCAL;

if (!uri) {
  throw new Error("❌ No MongoDB URI configured in .env");
}

const client = new MongoClient(uri);

await client.connect();
console.log(`✅ Connected to MongoDB (${mongoEnv})`);

// Expose db object ready to use
const db = client.db("mydb");

export { db };

File: api/database/mongodb.py
import os
from dotenv import load_dotenv
from pymongo import MongoClient

load_dotenv()

mongo_env = os.getenv("MONGO_ENV", "local")

uri = (
    os.getenv("MONGODB_URI_ATLAS")
    if mongo_env == "atlas"
    else os.getenv("MONGODB_URI_LOCAL")
)

if not uri:
    raise RuntimeError("❌ No MongoDB URI configured in .env")

client = MongoClient(uri)
db = client["mydb"]

print(f"✅ Connected to MongoDB ({mongo_env})")
